mvpa2/algorithms/statis.py

```diff
@@ -0,0 +1,142 @@
+    return z, np.dot(z, z.T)
+
+
+def get_eig(h):
+    """
+    Helper function to return sorted and sign flipped
+    Eigen decomposition.
+    :param C: Symmetric matrix
+    :return: eigen values, eigen vectors
+    """
+    evals, evecs = np.linalg.eigh(h)
+    # To deal with numerical errors
+    evals[evals<0] = 0
+    evecs = evecs[:, np.argsort(evals)[-1::-1]]
+    evals = np.sort(evals)[-1::-1]
+    # Fix signs
+    evecs = evecs*np.sign(evecs[0, 0])
+    return evals, evecs
+
+
+class Statis(ClassWithCollections):
+    """Implementation of STATIS as presented in:
+    Abdi, H., Williams, L.J., Valentin, D., & Bennani-Dosse, M. (2012). STATIS
+    and DISTATIS: Optimum multi-table principal component analysis and three way
+    metric multidimensional scaling. Wiley Interdisciplinary Reviews:
+    Computational Statistics, 4, 124-167.
+    """
+    tables_attr = 'chunks'
+    bootstrap_iter = 0
+
+    def __init__(self, tables_attr='chunks', **kwargs):
+        ClassWithCollections.__init__(self, **kwargs)
+        self.tables_attr = tables_attr
+        # Subject factor scores
+        self.G_t = None
+        # Sample factor scores
+        self.G_s = None
+        # output space dimensionality
+        self.outdim = None
+        # Subject specific weights for compromise matrix
+        self.alpha = None
+
+    def __call__(self, dss):
+        """
+        Derive compromise matrix and aligned space.
+        :param dss: a list of datasets
+        :return:
+        """
+        if type(dss) == list:
+            print("Processing each of %d list items as tables"%(len(dss)))
+            nss = [sd.nsamples for sd in dss]
+            if not nss.count(nss[0]) == len(nss):
+                raise ValueError, "All the datasets in the list should have matching" \
+                                "sample size. Input samples sizes are: %s" %nss
+        elif is_datasetlike(dss):
+            print("Processing each unique %s as table"%(self.tables_attr))
+            dss = [dss[:, dss.fa[self.tables_attr] == attr] for attr in
+                                    np.unique(dss.fa[self.tables_attr])]
+        else:
+            raise ValueError, "Input should either be a list of datasets/matrices" \
+                             " or a dataset"
+        # Normalizing data tables
+        dss = [normalize_ds(sd) for sd in dss]
+        nsamples = dss[0].nsamples
+        nfs = [sd.nfeatures for sd in dss]
+        self.outdim = min(nsamples-1, min(nfs))
+        # Deriving compromise matrix
+        cpms, C = get_Rv(dss)
+        e_t, ev_t = get_eig(C)
+        # XXX We can repeat the above two steps with permuted samples per column in each dataset
+        # to get a null distr of eigen values to evaluate how many are significant.
+        # First one should be, if second one is also, then there are two clusters.
+
+        # Factor scores for subjects/tables
+        self.G_t = np.dot(ev_t, np.diag(np.sqrt(e_t)))
```

Comment:
in DISTATIS, computing the compromise only makes sense if all pair-wise correlations between matrices from individuals are positively correlated. I don't know if that is also the case for STATIS... Is this something we may want to check, and throw an error if any pair is not positively correlated?